From 1277024bebc4e6c9f6bd18980c90deb37e63ea54 Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Sun, 18 Oct 2020 17:31:00 +0300
Subject: [PATCH] Append attachments as files with UTF8 header

---
 subscriptions_mail.cron.inc | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/subscriptions_mail.cron.inc b/subscriptions_mail.cron.inc
index fb5712e..5285774 100644
--- a/subscriptions_mail.cron.inc
+++ b/subscriptions_mail.cron.inc
@@ -296,10 +296,32 @@ function _subscriptions_mail_send($module, $mailkey, $name, $to, $from, $uid, $s
     return NULL;
   }
 
+  $default_from = variable_get('site_mail', ini_get('sendmail_from'));
+  $site_name = variable_get('site_name', 'Drupal');
+  $node = node_load($data['subs']['object']->nid);
+  $sender = user_load($node->uid);
+  if ($sender->realname) {
+    $from = '"' . mb_encode_mimeheader($sender->realname) . ' (via '. $site_name .')" <no-reply@internt.specped.su.se>';
+  } else {
+    $from = '"'.$sender->name.' (via '.$site_name.')" <no-reply@internt.specped.su.se>';
+  }
+
+  $attachments = array();
+  $language = field_language('node', $node, 'field_attachment');
+  if (isset($node->field_attachment[$language])) {
+    foreach($node->field_attachment[$language] as $file) {
+      $attachments[] = array(
+          'filename' => mb_encode_mimeheader($file['filename']),
+          'filepath' => $file['uri'],
+        );
+    }
+  }
+
   $mail_success = drupal_mail($module, $mailkey, $to, user_preferred_language($user), array(
     'data' => $data,
     'account' => $user,
     'object' => NULL,
+    'attachments' => $attachments,
   ), $from, TRUE);
 
   $to = check_plain($to);
-- 
2.28.0

