From 9010358163c87d79d15c3529137cd0688089daac Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Thu, 7 Mar 2019 20:47:11 +0300
Subject: [PATCH] Remove logout button from toolbar

---
 modules/toolbar/toolbar.module | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules/toolbar/toolbar.module b/modules/toolbar/toolbar.module
index 6d1b581d9f..7c3e18a205 100644
--- a/modules/toolbar/toolbar.module
+++ b/modules/toolbar/toolbar.module
@@ -220,10 +220,10 @@ function toolbar_view() {
         'html' => TRUE,
         'attributes' => array('title' => t('User account')),
       ),
-      'logout' => array(
-        'title' => t('Log out'),
-        'href' => 'user/logout',
-      ),
+  //      'logout' => array(
+  //      'title' => t('Log out'),
+  //      'href' => 'user/logout',
+  //    ),
     );
   }
   else {
-- 
2.21.0

