From 3bf1c41cdf4fb52a69d3fc921d3d0188fd7eb520 Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Tue, 7 Jul 2020 11:49:15 +0300
Subject: [PATCH] Use local branch for determining tag name

---
 git_deploy.module | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/git_deploy.module b/git_deploy.module
index 30a0bdc..64e61e2 100644
--- a/git_deploy.module
+++ b/git_deploy.module
@@ -254,6 +254,12 @@ function _git_deploy_get_upstream($git, array $patterns = array('*')) {
         $upstream['tag'] = $tag;
       }
     }
+    // If our local branch has a tag name higher than git describe one, use our tag
+    $branch = exec("$git rev-parse --abbrev-ref HEAD");
+    if (substr( $branch, 0, 4 ) === "tag_") {
+      $branchtag = substr($branch, 4);
+      $upstream['tag'] = ($branchtag > $tag) ? $branchtag : $tag;
+    }
     // Find the timestamp for the current commit.
     $upstream['datestamp'] = exec("$git log -1 --pretty=format:%at " . escapeshellarg($last_base) . " 2> " . GIT_DEPLOY_ERROR_DUMP);
   }
-- 
2.25.0

