From 9ba34233d70daa756e4d675af62597c7578441fe Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Sat, 8 Jun 2013 21:54:51 +0200
Subject: [PATCH] Hack for disabling buttons for reflection plugin

---
 mod/assign/renderer.php | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/mod/assign/renderer.php b/mod/assign/renderer.php
index b50ec931122..fa30a22af07 100644
--- a/mod/assign/renderer.php
+++ b/mod/assign/renderer.php
@@ -914,6 +914,22 @@ class mod_assign_renderer extends plugin_renderer_base {
         $o .= html_writer::table($t);
         $o .= $this->output->box_end();
 
+
+        // Hack for Reflection submission plugin.
+        if ($status->submission) {
+            global $CFG, $DB;
+            require_once($CFG->dirroot.'/group/lib.php');
+            $waitingid = $DB->get_field('assign_plugin_config', 'value', array('assignment' => $status->submission->assignment,
+                'plugin' => 'reflection', 'name' => 'waitingid'));
+            if ($waitingid) {
+                if (groups_is_member($waitingid, $status->submission->userid)) {
+                    $status->cansubmit = false;
+                    $status->canedit = false;
+                }
+            }
+        }
+
+
         // Links.
         if ($status->view == assign_submission_status::STUDENT_VIEW) {
             if ($status->canedit) {
-- 
2.24.0

