From ee68a3a101f0fea639793f993721e1fd71fafcd3 Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Fri, 1 Apr 2022 11:12:18 +0300
Subject: [PATCH] Hide badges menu link from the navigat

---
 course/lib.php | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/course/lib.php b/course/lib.php
index 803d28da3ac..5b51ea2ed32 100644
--- a/course/lib.php
+++ b/course/lib.php
@@ -3794,7 +3794,7 @@ function course_get_tagged_course_modules($tag, $exclusivemode = false, $fromcon
  * @since  Moodle 3.2
  */
 function course_get_user_navigation_options($context, $course = null) {
-    global $CFG, $USER;
+    global $CFG, $USER, $DB;
 
     $isloggedin = isloggedin();
     $isguestuser = isguestuser();
@@ -3830,7 +3830,8 @@ function course_get_user_navigation_options($context, $course = null) {
     if ($isfrontpage) {
         // We are on the front page, so make sure we use the proper capability (site:viewparticipants).
         $options->participants = course_can_view_participants($sitecontext);
-        $options->badges = !empty($CFG->enablebadges) && has_capability('moodle/badges:viewbadges', $sitecontext);
+        $options->badges = !empty($CFG->enablebadges) && has_capability('moodle/badges:viewbadges', $sitecontext) &&
++                            $DB->record_exists('badge', array('courseid' => $course->id));
         $options->tags = !empty($CFG->usetags) && $isloggedin;
         $options->search = !empty($CFG->enableglobalsearch) && has_capability('moodle/search:query', $sitecontext);
         $options->calendar = $isloggedin;
-- 
2.32.0 (Apple Git-132)

