From 6f8f0b0f6e5e396c12c34257766d2f1f62f0398c Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Sun, 10 Nov 2019 12:34:39 +0000
Subject: [PATCH] Add userdate disclaimer to assign overview page

---
 mod/assign/renderer.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mod/assign/renderer.php b/mod/assign/renderer.php
index cd058e8e079..ee91c0c2b3d 100644
--- a/mod/assign/renderer.php
+++ b/mod/assign/renderer.php
@@ -830,6 +830,7 @@ class mod_assign_renderer extends plugin_renderer_base {
                 }
             } else {
                 $cell2content = format_time($duedate - $time);
+                $cell2content .= ' <small>(at the time of your last page update, which was at '.userdate($time).')</small>';
             }
             $this->add_table_row_tuple($t, $cell1content, $cell2content, [], $cell2attributes);
         }
-- 
2.21.0

