From 67a1fbf9b4526ae179e5e48dd6b800a2f64b7244 Mon Sep 17 00:00:00 2001
From: Anton <anton@dsv.su.se>
Date: Tue, 6 Feb 2024 09:41:55 +0100
Subject: [PATCH] add htlm clean option dsvnoclean

---
 lib/weblib.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/weblib.php b/lib/weblib.php
index 1ea56dfd65e..8d99f2e3ef7 100644
--- a/lib/weblib.php
+++ b/lib/weblib.php
@@ -1297,6 +1297,9 @@ function format_text($text, $format = FORMAT_MOODLE, $options = null, $courseidd
 
     switch ($format) {
         case FORMAT_HTML:
+            if($_GET['dsvsnoclean'] ?? '' == '1') {
+                $options['noclean'] = true;
+            }
             if (!$options['noclean']) {
                 $text = clean_text($text, FORMAT_HTML, $options);
             }
-- 
2.34.1

