From b9e376c06bc09e16f81c6ba3dc0300c36e6c26ec Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Mon, 22 May 2017 15:23:49 +0200
Subject: [PATCH] Hide badges menu link from the navigaton

---
 course/lib.php | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/course/lib.php b/course/lib.php
index e60bd2c48b..3c324f908a 100644
--- a/course/lib.php
+++ b/course/lib.php
@@ -3779,7 +3779,7 @@ function course_get_tagged_course_modules($tag, $exclusivemode = false, $fromcon
  * @since  Moodle 3.2
  */
 function course_get_user_navigation_options($context, $course = null) {
-    global $CFG;
+    global $CFG, $DB;
 
     $isloggedin = isloggedin();
     $isguestuser = isguestuser();
@@ -3826,7 +3826,9 @@ function course_get_user_navigation_options($context, $course = null) {
     } else {
         $options->participants = has_capability('moodle/course:viewparticipants', $context);
         $options->badges = !empty($CFG->enablebadges) && !empty($CFG->badges_allowcoursebadges) &&
-                            has_capability('moodle/badges:viewbadges', $context);
+                            has_capability('moodle/badges:viewbadges', $context) &&
+                            $DB->record_exists('badge', array('courseid' => $course->id));
+
         // Add view grade report is permitted.
         $grades = false;
 
-- 
2.11.0 (Apple Git-81)

