From a553f60da3fc6fe4502548c3b524f43610f2f2d9 Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Tue, 19 Dec 2017 13:45:14 +0300
Subject: [PATCH] Add userdate disclaimer to assign overview page

---
 mod/assign/renderer.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mod/assign/renderer.php b/mod/assign/renderer.php
index 1e21ba2bdf..9975770d73 100644
--- a/mod/assign/renderer.php
+++ b/mod/assign/renderer.php
@@ -843,6 +843,7 @@ class mod_assign_renderer extends plugin_renderer_base {
                 }
             } else {
                 $cell2 = new html_table_cell(format_time($duedate - $time));
+                $cell2->text .= ' <small>(at the time of your last page update, which was at '.userdate($time).')</small>';
             }
             $row->cells = array($cell1, $cell2);
             $t->data[] = $row;
-- 
2.15.0

