From 9b5f5d0888b68ac329bb3498902a70ffa62f7e05 Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Thu, 16 Nov 2017 13:27:51 +0300
Subject: [PATCH] Display email when enrolling users

---
 enrol/externallib.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/enrol/externallib.php b/enrol/externallib.php
index ea5f971c9e..99a40baa1f 100644
--- a/enrol/externallib.php
+++ b/enrol/externallib.php
@@ -453,7 +453,7 @@ class core_enrol_external extends external_api {
                                                $params['perpage']);
 
         $results = array();
-        $requiredfields = ['id', 'fullname', 'profileimageurl', 'profileimageurlsmall'];
+        $requiredfields = ['id', 'fullname', 'email', 'profileimageurl', 'profileimageurlsmall'];
         foreach ($users['users'] as $id => $user) {
             // Note: We pass the course here to validate that the current user can at least view user details in this course.
             // The user we are looking at is not in this course yet though - but we only fetch the minimal set of
-- 
2.15.0

