From a5dd1b63e713f64e1a6b6d2c425e15b85629b009 Mon Sep 17 00:00:00 2001
From: Pavel Sokolov <pavel.m.sokolov@gmail.com>
Date: Wed, 11 Aug 2021 08:49:27 +0300
Subject: [PATCH] Move completion info to the right

---
 course/templates/activity_info.mustache | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/course/templates/activity_info.mustache b/course/templates/activity_info.mustache
index f415d2a4cf6..9073d0777b8 100644
--- a/course/templates/activity_info.mustache
+++ b/course/templates/activity_info.mustache
@@ -50,6 +50,11 @@
     }
 }}
 <div data-region="activity-information" data-activityname="{{activityname}}" class="activity-information">
+    {{^isautomatic}}
+        {{#showmanualcompletion}}
+            {{> core_course/completion_manual }}
+        {{/showmanualcompletion}}
+    {{/isautomatic}}
     {{#hasdates}}
     <div data-region="activity-dates" class="mb-1">
         {{#activitydates}}
@@ -67,11 +72,6 @@
                         {{/completiondetails}}
                     </div>
                 {{/isautomatic}}
-                {{^isautomatic}}
-                    {{#showmanualcompletion}}
-                        {{> core_course/completion_manual }}
-                    {{/showmanualcompletion}}
-                {{/isautomatic}}
             </div>
         {{/uservisible}}
     {{/hascompletion}}
-- 
2.32.0

