From 1f8aff9b18b1e7f14f831f0c39e610d488979ae3 Mon Sep 17 00:00:00 2001
From: orkit <ryan@dias.se>
Date: Wed, 14 Dec 2022 13:18:53 +0100
Subject: [PATCH] Downgrade updated version

---
 version.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/version.php b/version.php
index 0ffb464..66ffc81 100644
--- a/version.php
+++ b/version.php
@@ -29,7 +29,7 @@
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->version   = 2022051101;
-$plugin->requires  = 2022041200;
+$plugin->requires  = 2019051100;
 $plugin->component = 'mod_mediagallery';
 $plugin->maturity  = MATURITY_STABLE;
 $plugin->release   = '4.0.1';
-- 
2.38.1

