From f7e8841f72f9fe73fa837c275e2e7dba3c130319 Mon Sep 17 00:00:00 2001
From: orkit <ryan@dias.se>
Date: Wed, 14 Dec 2022 13:56:44 +0100
Subject: [PATCH] Temporary fix for Moodle 3 requirement

---
 version.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/version.php b/version.php
index 732509d..3ed4d6f 100644
--- a/version.php
+++ b/version.php
@@ -31,5 +31,5 @@ defined('MOODLE_INTERNAL') || die();
 $plugin->component = 'mod_scheduler'; // Full name of the plugin (used for diagnostics).
 $plugin->version   = 2022120200;      // The current module version (Date: YYYYMMDDXX).
 $plugin->release   = '4.x dev';       // Human-friendly version name.
-$plugin->requires  = 2022041900;      // Requires Moodle 4.0.
+$plugin->requires  = 2019052000;      // Requires Moodle 3.7.
 $plugin->maturity  = MATURITY_ALPHA;  // Development release - not for production use.
-- 
2.38.1

